contracts/stake/src/lib.rs

```diff
@@ -32,7 +20,65 @@ pub const MATURITY: u64 = 2 * EPOCH;
 /// The minimum amount of Dusk one can stake.
 pub const MINIMUM_STAKE: Dusk = dusk(1_000.0);
 
-pub const TX_STAKE: u8 = 0x00;
-pub const TX_UNSTAKE: u8 = 0x01;
-pub const TX_WITHDRAW: u8 = 0x02;
-pub const TX_ADD_ALLOWLIST: u8 = 0x03;
+extern crate alloc;
+
+pub mod contract;
+pub mod error;
+pub mod stake;
+
+pub use contract::StakeContract;
+pub use error::Error;
+pub use stake::Stake;
+
+#[cfg(target_family = "wasm")]
+#[path = ""]
+mod wasm {
```

Comment:
We should also consider a `stake-types` crate in the same style as `transfer-types`. This would include structs to use as arguments for the exports, but also functions to perform the necessary signatures - so we don't have to duplicate them in `dusk-wallet-core`.